catatonit.c

```diff
@@ -136,6 +139,64 @@ int kernel_signals[] = {SIGFPE, SIGILL, SIGSEGV, SIGBUS, SIGABRT, SIGTRAP, SIGSY
 
 #define ARRAY_LEN(arr) (sizeof(arr) / sizeof(*arr))
 
+static int syscall_close_range(unsigned int fd, unsigned int max_fd, unsigned int flags)
+{
+#ifdef __NR_close_range
+  return (int) syscall(__NR_close_range, fd, max_fd, flags);
+#else
+  errno = ENOSYS;
+  return -1;
+#endif
+}
+
+/*
+ * Close every fd >= n that is different from exclude_fd.
+ */
+static void close_fds_ge_than(int n, int exclude_fd)
+{
+	struct dirent *next;
+	DIR *dir;
+	int ret;
+	int fd;
+
+	if (exclude_fd < n
+	    && syscall_close_range(n, UINT_MAX, 0))
+		return;
+
+	if (exclude_fd == n
+	    && syscall_close_range(n + 1, UINT_MAX, 0))
+		return;
+
+	if (exclude_fd > n
+	    && syscall_close_range(n, exclude_fd - 1, 0) == 0
+	    && syscall_close_range(exclude_fd + 1, UINT_MAX, 0))
```

Comment:
This fails (overflows) if exclude_fd == UINT_MAX

Reply:
I've moved the close_range handling to a separate function